File: binary_search_algo/python_bin_search/iterative_method.py

def binarySearch(arr, target):
    high = len(arr)-1
    low = 0
    while low < high:
        mid = high + low // 2
        if(x == arr[mid]):
            return mid
        if(x < arr[mid]):
            high = mid-1
        if(x > arr[mid]):
            low = mid+1
           
    return -1
    

# Driver Code
arr = [2, 3, 4, 10, 40]
x = 3
 
# Function call
result = binarySearch(arr, x)
 
if result != -1:
    print("Element is present at index % d" % result)
else:
    print("Element is not present in array")

File: binary_search_algo/cpp_bin_search/iterative_method.cpp
#include <stdio.h> 
int binarySearch(int target, int* array, int high, int low);
int main(void) {

    int array[] = {1,2,3,4,5};
    int low = 0;
    int high = sizeof(array) / sizeof(*array);
    int result = binarySearch(5, array, high, low);
    
    printf("The target is at: %i\n", result);

}

int binarySearch(int target, int* array, int high, int low) {
    

    while(low < high) {
        int middle = low + high / 2;
        
        if(target == array[middle]) return middle;
        if(target < array[middle]) high = middle-1;
        else low = middle +1;
    }
    return -1;
}

